clients/lsp-javascript.el

```diff
@@ -44,7 +44,7 @@
 
 (defun lsp-typescript-javascript-tsx-jsx-activate-p (filename &optional _)
   "Check if the javascript-typescript language server should be enabled based on FILENAME."
-  (or (string-match-p (rx (one-or-more anything) "." (or "ts" "js") (opt "x") string-end) filename)
+  (or (string-match-p "\\`mjs\\|[jt]sx?\\'" (or (file-name-extension filename) ""))
```

Comment:
Can you please not change this more than needed? The `rx` expression was simpler and more maintainable. The `(one-or-more anything)` can be dropped, though:
```suggestion
  (or (string-match-p (rx "." (or "mjs" (: (or "ts" "js") (opt "x"))) string-end) filename)
```

Reply:
Nobody speaks rx, and it offers no performance benefits since most functions that accepts regular expressions accepts a string. Regular expression is annoying enough, Emacs regular expression even more so, and now with rx, you are asking the reader to translate what they want to do 3 times in their head before arriving to a syntax (does any know what ":"  is without RTFM?). Rx only makes sense If you have a really long and complicated regular expression with lots of branches and groups that you want to compose together with functions. There are only 5 cases here that you'll likely never change again. This client and eslint are the only clients in LSP that uses rx, neither have a compelling reason to do so.

Just keep it simple.

Reply:
Additional benefit to using my solution - you can test it in re-builder.